pkg/views/workspace/info/view.go

```diff
@@ -3,107 +3,146 @@
+	output += "\n"
+	output += getInfoLine(nameLabel, *workspace.Name) + "\n"
+
+	if isCreationView {
+		output += getInfoLine("Editor", ide) + "\n"
 	} else {
-		repoBranch = "Branch" + repositoryBranchStyle.Render(*project.Repository.Branch)
+		output += getInfoLine("ID", *workspace.Id) + "\n"
 	}
 
-	repoView := "Url" + (repositoryURLStyle.Render(*project.Repository.Url)) + "\n" + repoBranch
-	repoView = "Repository" + viewStyle.Render(repoView)
+	if len(workspace.Projects) == 1 {
+		output += renderSingleProject(&workspace.Projects[0], isCreationView)
+	} else {
+		output += renderProjects(workspace.Projects, isCreationView)
+	}
 
-	projectView := "Project" + projectNameStyle.Render(*projectInfo.Name) + "\n" + "State  " + projectState + extensionsTable
+	terminalWidth, _, err := term.GetSize(int(os.Stdout.Fd()))
+	if err != nil {
+		fmt.Println(output)
+		return
+	}
+	if terminalWidth < views.TUITableMinimumWidth || forceUnstyled {
+		renderUnstyledInfo(output)
+		return
+	}
 
-	return viewStyle.Render(projectView) + viewStyle.Render(repoView)
+	if !isCreationView {
+		output = views.GetStyledMainTitle("Workspace Info") + "\n" + output
+	}
+
+	renderTUIView(output, views.GetContainerBreakpointWidth(terminalWidth), isCreationView)
 }
 
-func Render(workspace *serverapiclient.WorkspaceDTO) {
-	if workspace == nil {
-		return
+func renderUnstyledInfo(output string) {
+	fmt.Println(output)
+}
+
+func renderTUIView(output string, width int, isCreationView bool) {
+	output = lipgloss.NewStyle().PaddingLeft(3).Render(output)
+
+	content := lipgloss.
+		NewStyle().Width(width).
+		Render(output)
+
+	if !isCreationView {
+		content = lipgloss.NewStyle().Margin(1, 0).Render(content)
 	}
 
+	fmt.Println(content)
+}
+
+func renderSingleProject(project *serverapiclient.Project, isCreationView bool) string {
 	var output string
-	output = "\n"
-	output += "Workspace" + workspaceNameStyle.Render(*workspace.Info.Name) + "\n"
-	output += "ID" + workspaceNameStyle.Render(*workspace.Id) + "\n"
-	output += "Target" + workspaceNameStyle.Render(*workspace.Target) + "\n"
 
-	if len(workspace.Projects) > 1 {
-		output += "\n" + "Projects"
+	repositoryUrl := *project.Repository.Url
+	repositoryUrl = strings.TrimPrefix(repositoryUrl, "https://")
+	repositoryUrl = strings.TrimPrefix(repositoryUrl, "http://")
+
+	output += getInfoLineState("State", project.State) + "\n"
+	if project.Target != nil && !isCreationView {
+		output += getInfoLine("Target", *project.Target) + "\n"
 	}
+	output += getInfoLine("Repository", repositoryUrl)
 
-	for _, project := range workspace.Projects {
-		for _, projectInfo := range workspace.Info.Projects {
-			if *projectInfo.Name == *project.Name {
-				output += projectRender(&projectInfo, &project)
-				break
-			}
+	if project.Name != nil && !isCreationView {
+		output += "\n"
+		output += getInfoLine("Project", *project.Name)
+	}
+
+	return output
+}
+
+func renderProjects(projects []serverapiclient.Project, isCreationView bool) string {
```

Comment:
`render...` functions should actually render, not return strings. We should use `get...` for functions that return views that should be rendered by something else.

(From what I can see, it's mostly consistent with that practice. These functions here are the outliers)